### PR TITLE
Added loop logic for starting gsma flow

### PR DESCRIPTION
**Pull request by ankita10r**
## Description

* Added the logic for looping in order to start bulk gsma flows

 _(Ignore if these details are present on the associated JIRA ticket)_

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [X] Design related bullet points or design document link related to this PR added in the description above. 

- [X] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [x] Create/update unit or integration tests for verifying the changes made.

- [X} Add required Swagger annotation and update API documentation with details of any API changes if applicable

- [X] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

@fynmanoj : Please review this. 
@danishjamal104 : for your reference

### CODE REVIEW

File: src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java [hunk 1]
```diff
@@ -98,14 +98,17 @@ public void configure() throws Exception {
...
-                    while(transactionList.size() > 0) {
-                        GSMATransaction gsmaTransaction = convertTxnToGSMA(transactionList.get(0));
-                        exchange.setProperty(GSMA_CHANNEL_REQUEST, gsmaTransaction);
-                        exchange.setProperty(INIT_SUB_BATCH_FAILED, false);
-                        transactionList.remove(0);
-                    }
+                    exchange.setProperty("length",transactionList.size());
 
                 })
+                .loop(simple("${exchangeProperty.length}"))
```
Comment (danishjamal104) [inline src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java:104]: Will this create a iteration `length` number of times??

File: src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java [hunk 2]
```diff
@@ -114,6 +117,7 @@ public void configure() throws Exception {
                 .marshal().json()
                 .to("direct:external-api-call") // loads the endpoint based on configuration and calls the external api
                 .log("Completed start of workflow for gsma")
+                .end()
```
Comment (danishjamal104) [inline src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java:120]: this `end` is for loop right??

Comment (ankita10r) [reply src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java:120]: yes

File: src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java (hunk 1 above)
Comment (ankita10r) [reply src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java:104]: yes, and length will be picked from this: `exchange.setProperty("length",transactionList.size())` on line 101

**Review by danishjamal104 — APPROVED** [review]
LGTM

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java [hunk 3]
```diff
@@ -98,14 +98,17 @@ public void configure() throws Exception {
...
-                        exchange.setProperty(GSMA_CHANNEL_REQUEST, gsmaTransaction);
-                        exchange.setProperty(INIT_SUB_BATCH_FAILED, false);
-                        transactionList.remove(0);
-                    }
+                    exchange.setProperty("length",transactionList.size());
 
                 })
+                .loop(simple("${exchangeProperty.length}"))
+                .process(exchange -> {
+                    logger.info("Loop index {}", exchange.getProperty("CamelLoopIndex", Integer.class));
```
[the remark this answers is not in the document]
Comment (ankita10r) [reply src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java:106]: ok

File: src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java [hunk 4]
```diff
@@ -98,14 +98,17 @@ public void configure() throws Exception {
...
+                    GSMATransaction gsmaTransaction = convertTxnToGSMA((Transaction) exchange.getProperty(TRANSACTION_LIST, List.class)
+                            .get(exchange.getProperty("CamelLoopIndex", Integer.class)));
```
[the remark this answers is not in the document]
Comment (ankita10r) [reply src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java:108]: ok

File: src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java [hunk 5]
```diff
@@ -98,14 +98,19 @@ public void configure() throws Exception {
...
-                        exchange.setProperty(GSMA_CHANNEL_REQUEST, gsmaTransaction);
-                        exchange.setProperty(INIT_SUB_BATCH_FAILED, false);
-                        transactionList.remove(0);
-                    }
+                    exchange.setProperty("length",transactionList.size());
 
                 })
+                .loop(simple("${exchangeProperty.length}"))
+                .process(exchange -> {
+                    exchange.setProperty(LOOP_INDEX, exchange.getProperty("CamelLoopIndex", Integer.class));
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java:106]: this is wrong.  LOOP INDEX already exist in Exchange. you can use that. 

to improve readability please assign this (`exchange.getProperty("CamelLoopIndex", Integer.class)`) in a local variable and use that while accessing the elements in the list

**Review by fynmanoj — CHANGES_REQUESTED** [review]

**Review by fynmanoj — APPROVED** [review]

**Pull request merged**